packages/react/src/components/TreeView/TreeView-test.js

```diff
@@ -1,95 +1,219 @@
+      const onSelectSpy = jest.fn();
+
+      render(
+        <TreeView label="Tree View">
           <TreeNode
-            renderIcon={Folder}
-            id="5-3"
-            value="5-3"
-            label="5-3"
-            isExpanded>
-            <TreeNode renderIcon={Document} id="5-4" value="5-4" label="5-4" />
-          </TreeNode>
-        </TreeNode>
-      </TreeView>
-    );
-    expect(wrapper).toMatchSnapshot();
-  });
+            onSelect={onSelectSpy}
+            id="Node 1"
+            data-testid="Node 1"
+            label="Node 1"
+          />
+        </TreeView>
+      );
 
-  it('should be able to hide the label', () => {
-    expect(wrapper.find('TreeLabel').text()).toBe('Tree view');
-    wrapper.setProps({ hideLabel: true });
-    expect(wrapper.find('TreeLabel').text()).toBeFalsy();
-  });
+      fireEvent.click(screen.getByTestId('Node 1'));
 
-  describe('Single node selection', () => {
-    it('should be able to preselect a node', () => {
-      expect(wrapper.find(`.${prefix}--tree-node--selected`).text()).toBe('1');
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(onSelectSpy).toHaveBeenCalled();
     });
 
-    it('should handle selection at the tree level', () => {
-      const onTreeSelect = jest.fn();
-      wrapper.setProps({ onSelect: onTreeSelect });
-      wrapper.find('TreeNode[value="2"]').simulate('click');
-      expect(onTreeSelect).toHaveBeenCalledTimes(1);
+    it('should respect selected prop', () => {
+      render(
+        <TreeView label="Tree View" selected={['Node 1']}>
+          <TreeNode id="Node 1" data-testid="Node 1" label="Node 1" />
+        </TreeView>
+      );
+
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(screen.getByRole('treeitem', { name: 'Node 1' })).toHaveClass(
+        'cds--tree-node--selected'
+      );
     });
 
-    it('should handle selection at the node level', () => {
-      wrapper.find('TreeNode[value="2"]').simulate('click');
-      expect(onTreeSelect).toHaveBeenCalledTimes(1);
-      expect(onNodeSelect).toHaveBeenCalledTimes(1);
+    it('should respect size prop', () => {
+      render(<TreeView size="xs" label="Tree" />);
+
+      const tree = screen.getByRole('tree');
+
+      expect(tree).toHaveClass('cds--tree--xs');
     });
   });
 
-  describe('Tree node expansion', () => {
-    it('Caret icon should not render in leaf nodes', () => {
-      expect(wrapper.find(CaretDown).length).toBe(2);
+  describe('behaves as expected', () => {
+    it('should render tree with two nodes', () => {
+      render(
+        <TreeView label="Tree View">
+          <TreeNode label="Node 1" />
+          <TreeNode label="Node 2" />
+        </TreeView>
+      );
+
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(screen.getByText('Node 1')).toBeInTheDocument();
+      expect(screen.getByText('Node 2')).toBeInTheDocument();
+    });
```

Comment:
This feels like a duplicate of `should render children as expected`, do you think so or no?

```suggestion
```